glx/glxcmds.c

```diff
@@ -893,97 +893,85 @@ __glXDisp_GetVisualConfigs(__GLXclientState * cl, GLbyte * pc)
-    int p, i, err;
+    int err;
 
     if (!validGlxScreen(cl->client, req->screen, &pGlxScreen, &err))
         return err;
 
     xGLXGetVisualConfigsReply reply = {
-        .type = X_Reply,
-        .sequenceNumber = client->sequence,
-        .length = (pGlxScreen->numVisuals *
-                   __GLX_SIZE_CARD32 * GLX_VIS_CONFIG_TOTAL) >> 2,
         .numVisuals = pGlxScreen->numVisuals,
         .numProps = GLX_VIS_CONFIG_TOTAL
     };
 
     if (client->swapped) {
-        swaps(&reply.sequenceNumber);
-        swapl(&reply.length);
         swapl(&reply.numVisuals);
         swapl(&reply.numProps);
     }
 
-    WriteToClient(client, sizeof(xGLXGetVisualConfigsReply), &reply);
+    x_rpcbuf_t rpcbuf = { .swapped = client->swapped, .err_clear = TRUE };
 
-    for (i = 0; i < pGlxScreen->numVisuals; i++) {
+    for (int i = 0; i < pGlxScreen->numVisuals; i++) {
         modes = pGlxScreen->visuals[i];
 
-        p = 0;
-        buf[p++] = modes->visualID;
-        buf[p++] = glxConvertToXVisualType(modes->visualType);
-        buf[p++] = (modes->renderType & GLX_RGBA_BIT) ? GL_TRUE : GL_FALSE;
-
-        buf[p++] = modes->redBits;
-        buf[p++] = modes->greenBits;
-        buf[p++] = modes->blueBits;
-        buf[p++] = modes->alphaBits;
-        buf[p++] = modes->accumRedBits;
-        buf[p++] = modes->accumGreenBits;
-        buf[p++] = modes->accumBlueBits;
-        buf[p++] = modes->accumAlphaBits;
-
-        buf[p++] = modes->doubleBufferMode;
-        buf[p++] = modes->stereoMode;
-
-        buf[p++] = modes->rgbBits;
-        buf[p++] = modes->depthBits;
-        buf[p++] = modes->stencilBits;
-        buf[p++] = modes->numAuxBuffers;
-        buf[p++] = modes->level;
-
-        assert(p == GLX_VIS_CONFIG_UNPAIRED);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->visualID);
+        x_rpcbuf_write_CARD32(&rpcbuf, glxConvertToXVisualType(modes->visualType));
+        x_rpcbuf_write_CARD32(&rpcbuf, (modes->renderType & GLX_RGBA_BIT) ? GL_TRUE : GL_FALSE);
+
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->redBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->greenBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->blueBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->alphaBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->accumRedBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->accumGreenBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->accumBlueBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->accumAlphaBits);
+
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->doubleBufferMode);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->stereoMode);
+
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->rgbBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->depthBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->stencilBits);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->numAuxBuffers);
+        x_rpcbuf_write_CARD32(&rpcbuf, modes->level);
+
+        assert((rpcbuf.wpos / sizeof(CARD32)) == GLX_VIS_CONFIG_UNPAIRED);
```

Comment:
This will work for the first iteration (i = 0), but what about the following iterations when wpos will keep growing? Same question for other expressions below involving `rpcbuf.wpos`.

Reply:
do we need this assert() at all ?